mjpc/planners/dr_sampling/planner.cc

```diff
@@ -292,40 +299,73 @@ void DRSamplingPlanner::Rollouts(int num_trajectory, int horizon,
 
   policy.num_parameters = model->nu * policy.num_spline_points;
 
-  // random search
+  // compute num_trajectory random control tapes, storing each one in
+  // this->candidate_policy[i]. Additional copies of each tape are stored in
+  // this->candidate_policy[i + j*num_trajectory] for j=1,...,kNumRandomizedModels
   int count_before = pool.GetCount();
   for (int i = 0; i < num_trajectory; i++) {
-    pool.Schedule([&s = *this, &model = this->model, &task = this->task,
-                   &state = this->state, &time = this->time,
-                   &mocap = this->mocap, &userdata = this->userdata, horizon,
-                   i]() {
+    pool.Schedule([&s = *this, i, num_trajectory]() {
       // copy nominal policy
       {
         const std::shared_lock<std::shared_mutex> lock(s.mtx_);
         s.candidate_policy[i].CopyFrom(s.policy, s.policy.num_spline_points);
         s.candidate_policy[i].representation = s.policy.representation;
       }
 
-      // sample noise policy
+      // add random noise to the policy
       if (i != 0) s.AddNoiseToPolicy(i);
 
-      // ----- rollout sample policy ----- //
-
-      // policy
-      auto sample_policy_i = [&candidate_policy = s.candidate_policy, &i](
-                                 double* action, const double* state,
-                                 double time) {
-        candidate_policy[i].Action(action, state, time);
-      };
-
-      // policy rollout
-      s.trajectory[i].Rollout(
-          sample_policy_i, task, model, s.data_[ThreadPool::WorkerId()].get(),
-          state.data(), time, mocap.data(), userdata.data(), horizon);
+      // make copies of the candidate policy for each randomized model
+      for (int j = 1; j < kNumRandomizedModels; j++) {
+        int k = i + j * num_trajectory;
+        s.candidate_policy[k].CopyFrom(
+            s.candidate_policy[i], s.candidate_policy[i].num_spline_points);
+        s.candidate_policy[k].representation =
+            s.candidate_policy[i].representation;
+      }
     });
   }
   pool.WaitCount(count_before + num_trajectory);
   pool.ResetCount();
+  
+  // Toll out the control tapes across the randomized models. Tape i for model j
+  // is stored in this->trajectory[i + j*num_trajectory]. 
+  count_before = pool.GetCount();
+  for (int i=0; i < num_trajectory; i++) {
```

Comment:
if i'm understanding this correctly, we should maybe change the name of this variable to something like `num_trajectory_per_model` (or something shorter)

Reply:
yup, I agree that it's confusing terminology. Just changed to `num_rollouts` to match the GUI params - I'm open to other suggestions too, it's easy to change

Reply:
that seems good to me! we should also add some guardrails so that `num_rollouts * num_models` doesn't exceed `kMaxNumTrajectories` or whatever that variable is called.

Reply:
I just added a GUI parameter for the number of models. I've got guard rails on `num_rollouts` and `num_models` separately, which I think should do the trick but a second set of eyes on that would be helpful